### PR TITLE
Settings: adjust copy size and spacing

### PR DESCRIPTION
**Pull request by crisbusquets**
## Proposed changes:
This pull request includes a series of changes aimed at improving the visual interface, hierarchy and spacing within Jetpack Settings:

* Font-size adjustment: Increased the font size from 13px to 14px across various elements to enhance readability and ensure consistency throughout the application.

* Spacing adjustment: Removed unnecessary spacing and introduced appropriate spacing adjustments, particularly between radio buttons, to optimize visual balance.

Below are some examples:

| Before | After |
|-----------------|-----------------|
| Screenshot 2024-05-16 at 14 16 10 | Screenshot 2024-05-16 at 14 16 03 |
| Screenshot 2024-05-16 at 14 17 07 | Screenshot 2024-05-16 at 14 17 02 |
| Screenshot 2024-05-16 at 14 18 01 | Screenshot 2024-05-16 at 14 18 15 |

## Testing instructions:
* In Jetpack Settings, check that font-sizes look the same (14px), and that there's no 13px size anywhere.

### CODE REVIEW

File: .vscode/settings.json
Comment (keoshi) [inline .vscode/settings.json:1]: Unsure we want to add this to the shared VS Code settings. An engineer should weigh in here.

File: projects/plugins/jetpack/_inc/client/settings/style.scss
Comment (keoshi) [inline projects/plugins/jetpack/_inc/client/settings/style.scss:44]: Should we always use `rem()` to define sizing?

File: projects/plugins/jetpack/_inc/client/settings/style.scss
Comment (keoshi) [inline projects/plugins/jetpack/_inc/client/settings/style.scss:73]: Same as the comment above.

**Review by keoshi — DISMISSED** [review]
LGTM! Nicely done, Cris!

Left a couple of comments that aren't blockers, and also want to ask if we should move the `<span className="jp-form-toggle-explanation">` inside the `ModuleToggle`,  `renderToggle`, and `RadioControl` if that's the markup we need to reuse on all of them.

**Comment by keoshi** [comment]
Also re-ran the `Critical CSS e2e tests` since they were failing, but seems like the PR needs to be rebased before merging.

File: projects/plugins/jetpack/_inc/client/settings/style.scss
Comment (crisbusquets) [reply projects/plugins/jetpack/_inc/client/settings/style.scss:73]: The current code confuses me a bit, because sometimes we use `rem( 16px )` and sometimes we don't. In this case and the above I decided to follow the original convention.

File: .vscode/settings.json
Comment (crisbusquets) [reply .vscode/settings.json:1]: Probably not 😅

File: projects/plugins/jetpack/_inc/client/settings/style.scss
Comment (crisbusquets) [reply projects/plugins/jetpack/_inc/client/settings/style.scss:44]: Replied below: https://github.com/Automattic/jetpack/pull/37418#discussion_r1603424677

File: .vscode/settings.json
Comment (simison) [reply .vscode/settings.json:1]: Nope. :-)

**Review by lezama — APPROVED** [review]

**Pull request merged**
